src/BingAdsEventForwarder.js

```diff
@@ -180,10 +220,126 @@ var constructor = function() {
         return obj;
     }
 
+    function getEventConsentState(eventConsentState) {
+        return eventConsentState && eventConsentState.getGDPRConsentState
+            ? eventConsentState.getGDPRConsentState()
+            : {};
+    }
+
+    function generateConsentPayload(consentState, mappings) {
+        if (!mappings) {
+            return {};
+        }
+
+        var payload = cloneObject(self.consentPayloadDefaults);
+        if (mappings && mappings.length > 0) {
+            for (var i = 0; i < mappings.length; i++) {
+                var mappingEntry = mappings[i];
+                var mpMappedConsentName = mappingEntry.map.toLowerCase();
+                var bingMappedConsentName = mappingEntry.value;
+
+                if (
+                    consentState[mpMappedConsentName] &&
+                    bingConsentProperties.indexOf(bingMappedConsentName) !== -1
+                ) {
+                    payload[bingMappedConsentName] = consentState[
+                        mpMappedConsentName
+                    ].Consented
+                        ? bingConsentValues.Granted
+                        : bingConsentValues.Denied;
+                }
+            }
+        }
+
+        return payload;
+    }
+
+    function maybeSendConsentUpdateToBing(consentState) {
+        // If consent payload is empty,
+        // we never sent an initial default consent state
+        // so we shouldn't send an update.
```

Comment:
I think the initial default consent state will always have been sent based on us sending a default of granted even if they didn't select something in the UI, so this comment doesn't apply.